.github/scripts/Release.py

```diff
@@ -0,0 +1,580 @@
+
+    Args:
+      metadata: The project metadata read from the YAML file. This is the main
+        source of information for this function.
+      github: The GitHub API client. We use it to render the description to
+        HTML in a way that's consistent with GitHub's rendering.
+
+    Returns:
+      Metadata in the format that the Zenodo API expects.
+    """
+    # Generate the DOI author list from the authors in the project metadata
+    # TODO: Check the metadata file is consistent, maybe elsewhere:
+    zenodo_creators = []
+    for author_tier in ['Core', 'Developers', 'Contributors']:
+        for author in metadata['Authors'][author_tier]['List']:
+            zenodo_creator = dict(name=author['Name'])
+            if 'Orcid' in author:
+                zenodo_creator['orcid'] = author['Orcid']
+            if 'Affiliations' in author and len(author['Affiliations']) > 0:
+                zenodo_creator['affiliation'] = ' and '.join(
+                    author['Affiliations'])
+            zenodo_creators.append(zenodo_creator)
+    # Render the description to HTML
+    rendered_description = github.render_markdown_raw(metadata['Description'])
+    # Construct Zenodo metadata
+    return dict(title=metadata['Name'],
+                version=metadata['Version'],
+                publication_date=metadata['PublicationDate'],
+                doi=metadata['Doi'],
+                description=rendered_description,
+                creators=zenodo_creators,
+                related_identifiers=[
+                    dict(identifier=metadata['Homepage'],
+                         relation='isDocumentedBy',
+                         scheme='url',
+                         resource_type='publication-softwaredocumentation'),
+                    dict(identifier=('https://github.com/' +
+                                     metadata['GitHub']),
+                         relation='isSupplementTo',
+                         scheme='url',
+                         resource_type='software')
+                ],
+                language='eng',
+                communities=[dict(identifier='sxs')],
+                keywords=metadata['Keywords'],
+                license=metadata['License'],
+                upload_type='software',
+                access_right='open')
+
+
+def prepare(metadata: dict, version_name: str, metadata_file: str,
+            readme_file: str, zenodo: Zenodo, github: Github,
+            update_only: bool, check_only: bool):
+    # Validate new version name
+    match_version_name = re.match(VERSION_PATTERN + '$', version_name)
+    if not match_version_name:
+        raise ValueError(
+            "Version name '{}' doesn't match pattern '{}'.".format(
+                version_name, VERSION_PATTERN))
+    publication_date = '{}-{}-{}'.format(*match_version_name.groups()[:3])
+
+    if update_only:
+        # The new version already exists on Zenodo, and we assume the metadata
```

Comment:
Sorry, I don't really understand the difference between the if and else branch. I think this boils down to me not understanding what `update_only` means. What is being updated?

Reply:
This is the help string for `update_only`:

> Only update an existing version draft on Zenodo, not creating a new one. Use this flag if the metadata in the repository already reference the new version draft on Zenodo.

Reply:
Sorry, where is the help string located? I tried searching this file and couldn't find it.

Reply:
Okay, since I understand that message but neither of the comments related to the if-else could you reword the code comments? I think the thing that confuses me is `exists on Zenodo` sounds like we have hit a bug and should abort.

Reply:
reworded in a fixup